Lib/inspect.py

```diff
@@ -1242,6 +1242,14 @@ def getblock(lines):
             blockfinder.tokeneater(*_token)
     except (EndOfBlock, IndentationError):
         pass
+    except SyntaxError as e:
+        if "unmatched" not in e.msg:
```

Comment:
maybe also check `blockfinder.islambda` ?

Reply:
I want to be a bit more generic just in case this happens with other constructs.